pa-rserve/src/main/java/org/ow2/parserve/util/rsession/StartRserve.java

```diff
@@ -0,0 +1,253 @@
+                } else if (rsrvargs.contains("port")) {
+                    String rsport = rsrvargs.split("port")[1].trim().split("=")[1].trim().split(",")[0].trim();
+                    port = Integer.parseInt(rsport);
+                }
+                logger.info("Trying to connect to RServe on port : " + port);
+                if (rsrvargs.contains("--RS-port") || rsrvargs.contains("port")) {
+                    c = new RConnection("localhost", port);
+                } else {
+                    c = new RConnection("localhost");
+                }
+                logger.info("Rserve is running.");
+                c.close();
+                return true;
+            } catch (Exception e2) {
+                logger.debug("Try failed with: " + e2.getMessage());
+            }
+            /* a safety sleep just in case the start up is delayed or asynchronous */
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException ix) {
+            }
+
+            attempts--;
+        }
+        return false;
+    }
+
+    /**
+     * checks whether Rserve is running and if that's not the case it attempts to start it using the defaults for the platform where it is run on.
+     * This method is meant to be set-and-forget and cover most default setups. For special setups you may get more control over R with <code>launchRserve</code> instead.
+     */
+    public static boolean checkLocalRserve() {
+        if (isRserveRunning()) {
+            return true;
+        }
+        String osname = System.getProperty("os.name");
+        if (osname != null && osname.length() >= 7 && osname.substring(0, 7).equals("Windows")) {
+            logger.info("Windows: query registry to find where R is installed ...");
+            String installPath = null;
+            try {
+                Process rp = Runtime.getRuntime().exec("reg query HKLM\\Software\\R-core\\R");
+                Utils.RegistryHog regHog = new Utils.RegistryHog(rp.getInputStream(), true);
+                rp.waitFor();
+                regHog.join();
+                installPath = regHog.getInstallPath();
+            } catch (Exception rge) {
+                logger.error("ERROR: unable to run REG to find the location of R: " + rge);
+                return false;
+            }
+            if (installPath == null) {
+                logger.error("ERROR: canot find path to R. Make sure reg is available and R was installed with registry settings.");
+                return false;
+            }
+            return launchRserve(installPath + "\\bin\\R.exe");
+        }
+        return (launchRserve("R")
+                || /* try some common unix locations of R */ ((new File("/Library/Frameworks/R.framework/Resources/bin/R")).exists() && launchRserve("/Library/Frameworks/R.framework/Resources/bin/R"))
+                || ((new File("/usr/local/lib/R/bin/R")).exists() && launchRserve("/usr/local/lib/R/bin/R"))
```

Comment:
This file is from the authors of Rserve, I didn't want to modify it too much. On linux machine, usually the "R" command is directly available and we don't have to bother.